include/SDL3/SDL_render.h

```diff
@@ -227,7 +227,7 @@ extern DECLSPEC int SDLCALL SDL_CreateWindowAndRenderer(int width, int height, U
  * \sa SDL_GetRenderDriver
  * \sa SDL_GetRendererInfo
  */
-extern DECLSPEC SDL_Renderer * SDLCALL SDL_CreateRenderer(SDL_Window *window, const char *name, Uint32 flags);
+extern DECLSPEC SDL_Renderer * SDLCALL SDL_CreateRenderer(SDL_Window *window, const char *name, SDL_RendererFlags flags);
```

Comment:
This is an issue in c++ mode: or'ing 2 enums creates an int, not a enum value.
You'd need to override the `|` operator for it to work.

```
/home/maarten/projects/SDL/test/testcpp.cpp:13:85: error: invalid conversion from ‘int’ to ‘SDL_RendererFlags’ [-fpermissive]
   13 |     SDL_Renderer *renderer = SDL_CreateRenderer(window, NULL, SDL_RENDERER_SOFTWARE | SDL_RENDERER_PRESENTVSYNC);
      |                                                               ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                     |
      |                                                                                     int
In file included from /home/maarten/projects/SDL/include/SDL3/SDL.h:69,
                 from /home/maarten/projects/SDL/test/testcpp.cpp:1:
/home/maarten/projects/SDL/include/SDL3/SDL_render.h:230:115: note:   initializing argument 3 of ‘SDL_Renderer* SDL_CreateRenderer(SDL_Window*, const char*, SDL_RendererFlags)’
  230 | extern DECLSPEC SDL_Renderer * SDLCALL SDL_CreateRenderer(SDL_Window *window, const char *name, SDL_RendererFlags flags);
      |                                                                                                 ~~~~~~~~~~~~~~~~~~^~~~~
```

```c++
static SDL_RendererFlags operator|(SDL_RendererFlags lhs, SDL_RendererFlags rhs) {
    return (SDL_RendererFlags)((Uint32)lhs | (Uint32)rhs);
}
```

Reply:
> This is an issue in c++ mode: or'ing 2 enums creates an int, not a enum value.

And there may be even more problematic places. The whole enum thing should be avoided IMO, but that's me.

> You'd need to override the `|` operator for it to work.

Ugh, the abomination...

Reply:
I remember, but I can't find, a discussion where we chose to use Uint's instead of typedefs to keep the code somewhat readable. Or am I misremembering this?

Reply:
Vulkan solves this by creating a uint32_t typedef of VkFlags and then using that anywhere flags can be or'd together, e.g.
```c
typedef enum VkImageAspectFlagBits {
    VK_IMAGE_ASPECT_COLOR_BIT = 0x00000001,
    VK_IMAGE_ASPECT_DEPTH_BIT = 0x00000002,
    VK_IMAGE_ASPECT_STENCIL_BIT = 0x00000004,
    VK_IMAGE_ASPECT_METADATA_BIT = 0x00000008,
    VK_IMAGE_ASPECT_PLANE_0_BIT = 0x00000010,
    VK_IMAGE_ASPECT_PLANE_1_BIT = 0x00000020,
    VK_IMAGE_ASPECT_PLANE_2_BIT = 0x00000040,
    VK_IMAGE_ASPECT_NONE = 0,
    VK_IMAGE_ASPECT_MEMORY_PLANE_0_BIT_EXT = 0x00000080,
    VK_IMAGE_ASPECT_MEMORY_PLANE_1_BIT_EXT = 0x00000100,
    VK_IMAGE_ASPECT_MEMORY_PLANE_2_BIT_EXT = 0x00000200,
    VK_IMAGE_ASPECT_MEMORY_PLANE_3_BIT_EXT = 0x00000400,
    VK_IMAGE_ASPECT_PLANE_0_BIT_KHR = VK_IMAGE_ASPECT_PLANE_0_BIT,
    VK_IMAGE_ASPECT_PLANE_1_BIT_KHR = VK_IMAGE_ASPECT_PLANE_1_BIT,
    VK_IMAGE_ASPECT_PLANE_2_BIT_KHR = VK_IMAGE_ASPECT_PLANE_2_BIT,
    VK_IMAGE_ASPECT_NONE_KHR = VK_IMAGE_ASPECT_NONE,
    VK_IMAGE_ASPECT_FLAG_BITS_MAX_ENUM = 0x7FFFFFFF
} VkImageAspectFlagBits;
typedef VkFlags VkImageAspectFlags;
```
We should probably do something similar here.

Reply:
This PR doesn't exclusively deal with flags-type enums. I can revert flag changes for now and keep only the single value enum changes - those shouldn't cause issues in C++. You can continue this discussion in https://github.com/libsdl-org/SDL/issues/9186 in that case.